src/inet/InetInterface.cpp

```diff
@@ -445,7 +445,7 @@ CHIP_ERROR InterfaceId::GetInterfaceName(char * nameBuf, size_t nameBufSize) con
         {
             return CHIP_ERROR_POSIX(errno);
         }
-        size_t nameLength = strlen(intfName);
+        size_t nameLength = strnlen(intfName, IF_NAMESIZE);
```

Comment:
Again, the `CopyString` will just walk off the end of the data, no?  Worse yet, this can break the length check below and turn what would have been an out-of-bounds _read_ into an out-of-bounds _write_, which is much worse.

I'm stopping here.  The basic premise that strnlen is safer than strlen is only true in some cases, and so far it's not true in a bunch of cases in this PR.